### ISSUE TITLE
Mapping device local buffer on Metal panics

### ISSUE THREAD

**Issue by swiftcoder** [gfx-rs/gfx#2626]
Short info header:
- GFX version: gfx-backend-metal = "0.1.0"
- OS: Mac OS Mojave
- GPU: NVidia GT 650M

Attempting to perform device.map_memory() a memory object that was created as DEVICE_LOCAL panics under the metal backend, rather than returning `mapping::Error::InvalidAccess` (which seems like the most relevant error code).

Is this by design, or should we remove the panic in favour of the error return code?

**Comment by kvark** [comment]
Turning it into an error would be nice, but not hugely important. It's not like we expect anyone to handle this gracefully.

**Comment by swiftcoder** [comment]
Fair enough. It's a surprisingly easy situation to get into when you map the wrong buffer :)
